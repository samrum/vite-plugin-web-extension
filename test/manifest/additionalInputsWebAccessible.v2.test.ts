import { getResourceDir, runManifestV2Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("additionalInputsWebAccessible");

runManifestV2Test("additionalInputsWebAccessible", () => ({}), {
  additionalInputs: {
    scripts: [
      `${resourceDir}/script1.js`,
      {
        fileName: `${resourceDir}/script2.js`,
        webAccessible: false,
      },
      {
        fileName: `${resourceDir}/script3.js`,
        webAccessible: true,
      },
      {
        fileName: `${resourceDir}/script4.js`,
        webAccessible: {
          matches: ["https://example.com/"],
        },
      },
      {
        fileName: `${resourceDir}/script5.ts`,
        webAccessible: {
          matches: ["https://example.com/"],
          includeEntryFile: false,
        },
      },
      {
        fileName: `${resourceDir}/script6.js`,
        webAccessible: {
          extensionIds: ["oilkjaldkfjlasdf"],
        },
      },
      {
        fileName: `${resourceDir}/script7.js`,
        webAccessible: {
          extensionIds: ["oilkjaldkfjlasdf"],
          includeEntryFile: false,
        },
      },
      {
        fileName: `${resourceDir}/chunkedScript1.js`,
        webAccessible: {
          matches: ["https://example.com/"],
        },
      },
      {
        fileName: `${resourceDir}/chunkedScript2.js`,
        webAccessible: {
          matches: ["https://example.com/"],
          includeEntryFile: false,
        },
      },
    ],
  },
});
