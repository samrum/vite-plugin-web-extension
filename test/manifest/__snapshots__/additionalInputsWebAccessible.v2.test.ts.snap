// Vitest Snapshot v1

exports[`additionalInputsWebAccessible - Manifest V2 1`] = `
[
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"chunkedScript1\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/additionalInputsWebAccessible/chunkedScript1.js",
    "fileName": "assets/test/manifest/resources/additionalInputsWebAccessible/chunkedScript1.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/additionalInputsWebAccessible/chunkedScript1.js": {
        "code": "log(\\"chunkedScript1\\");",
        "originalLength": 57,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 22,
      },
    },
    "name": "test/manifest/resources/additionalInputsWebAccessible/chunkedScript1",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"chunkedScript2\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/additionalInputsWebAccessible/chunkedScript2.js",
    "fileName": "assets/test/manifest/resources/additionalInputsWebAccessible/chunkedScript2.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/additionalInputsWebAccessible/chunkedScript2.js": {
        "code": "log(\\"chunkedScript2\\");",
        "originalLength": 57,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 22,
      },
    },
    "name": "test/manifest/resources/additionalInputsWebAccessible/chunkedScript2",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"chunkedScript3\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/additionalInputsWebAccessible/chunkedScript3.js",
    "fileName": "assets/test/manifest/resources/additionalInputsWebAccessible/chunkedScript3.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/additionalInputsWebAccessible/chunkedScript3.js": {
        "code": "log(\\"chunkedScript3\\");",
        "originalLength": 57,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 22,
      },
    },
    "name": "test/manifest/resources/additionalInputsWebAccessible/chunkedScript3",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "function log(message) {
  console.log(message);
}
export {
  log as l
};
",
    "dynamicImports": [],
    "exports": [
      "l",
    ],
    "facadeModuleId": null,
    "fileName": "assets/log.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": false,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "log",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script1.js",
    "name": undefined,
    "source": "console.log(\\"script1\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script2.js",
    "name": undefined,
    "source": "console.log(\\"script2\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script3.js",
    "name": undefined,
    "source": "console.log(\\"script3\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script4.js",
    "name": undefined,
    "source": "console.log(\\"script4\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script5.js",
    "name": undefined,
    "source": "console.log(\\"script5\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/chunkedScript1.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/additionalInputsWebAccessible/chunkedScript1.js\\"))})();",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/chunkedScript2.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/additionalInputsWebAccessible/chunkedScript2.js\\"))})();",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/chunkedScript3.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/additionalInputsWebAccessible/chunkedScript3.js\\"))})();",
    "type": "asset",
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 2,
  \\"web_accessible_resources\\": [
    \\"test/manifest/resources/additionalInputsWebAccessible/script1.js\\",
    \\"test/manifest/resources/additionalInputsWebAccessible/script3.js\\",
    \\"test/manifest/resources/additionalInputsWebAccessible/script4.js\\",
    \\"test/manifest/resources/additionalInputsWebAccessible/script5.js\\",
    \\"assets/test/manifest/resources/additionalInputsWebAccessible/chunkedScript1.js\\",
    \\"assets/log.js\\",
    \\"test/manifest/resources/additionalInputsWebAccessible/chunkedScript1.js\\",
    \\"assets/test/manifest/resources/additionalInputsWebAccessible/chunkedScript2.js\\",
    \\"test/manifest/resources/additionalInputsWebAccessible/chunkedScript2.js\\",
    \\"assets/test/manifest/resources/additionalInputsWebAccessible/chunkedScript3.js\\",
    \\"test/manifest/resources/additionalInputsWebAccessible/chunkedScript3.js\\"
  ]
}",
    "type": "asset",
  },
]
`;
